Navigation 1775

paolodamico opened the pull request:
## Changes

Closes #1775. New navigation is behind the `navigation-1775` feature flag.

Also fixes some random styles that we missed on #2114 (Closes #2242).

https://github.com/PostHog/posthog.com/pull/501 has been updated to reflect the latest navigation items.

**Transitory.**
- As @timgl is working on more refactoring for actions (see #2046), this version keeps "Events" & "Actions" as separate menu items, but these should probably go together in a single view.
- This PR does not separate insights items (e.g. funnels or retention) out of the main "Insights" menu to be able to do this as a separate experiment.
- We're currently keeping the "Project" menu item, but as a separate experiment, we'll try to keep access to project settings only from the top bar and command palette.

paolodamico committed:
use antd with less & override styles

paolodamico committed:
override base ant design & prepare new styling

paolodamico committed:
removes legacy Flaticon icons/font

paolodamico committed:
removed legacy vars and updated base background

paolodamico committed:
restore white background as default

paolodamico committed:
introduce PageHeader & start cleaning up styles

paolodamico committed:
convert more pages

paolodamico committed:
convert all pages to new design

paolodamico committed:
remove /people namespace in url for cohorts & persons

paolodamico committed:
fixes to command palette

paolodamico committed:
remove bootstrap deps

paolodamico committed:
fix preflight

paolodamico committed:
refactor styles for localized-versions

paolodamico committed:
toast styles

paolodamico committed:
removes more unused components

paolodamico committed:
chop more bootstrap

paolodamico committed:
fix retention styles

paolodamico committed:
abstracted vars & events style cleanup

paolodamico committed:
removed unused funnel styles

paolodamico committed:
funnels

paolodamico committed:
insights

paolodamico committed:
action filter dropdown

paolodamico committed:
SendEventsOverlay

paolodamico committed:
fix build

paolodamico committed:
fix

paolodamico committed:
fix cypress

paolodamico committed:
remove bootstrap card

paolodamico committed:
remove custom card implementation

paolodamico committed:
remove more bootstrap

paolodamico committed:
get rid of bootstrap forms

paolodamico committed:
remove input groups

paolodamico committed:
fix cypress

paolodamico committed:
remove bootstrap grid

paolodamico committed:
remove badges & fix close

paolodamico committed:
keep bootstrap on static Django views

paolodamico committed:
remove button group I

paolodamico committed:
remove button group II

paolodamico committed:
remove button groups

paolodamico committed:
remove buttons & dropdown I (removes buggy tooltip)

paolodamico committed:
remove more dropdowns

paolodamico committed:
command palette styles

paolodamico committed:
remove bootstrap buttons

paolodamico committed:
remove dropdown

paolodamico committed:
remove bootstrap spacing

paolodamico committed:
completely remove bootstrap

paolodamico committed:
papercups branding & remove papercups on debug

paolodamico committed:
more style adjustments

paolodamico committed:
more styles

paolodamico committed:
graphs colors

paolodamico committed:
Merge branch 'master' into otterhog-ui

paolodamico committed:
fixes

paolodamico committed:
fixes

paolodamico committed:
more UI fixes

paolodamico committed:
remove card on white background pages

paolodamico committed:
fix settings

paolodamico committed:
Merge branch 'master' into otterhog-ui

paolodamico committed:
change ordering of fields in signup

paolodamico committed:
fix cypress

paolodamico committed:
fixes

paolodamico committed:
fix cypress

paolodamico committed:
new navigation menu with Ant

paolodamico committed:
convert to custom component that doesn't bounce weirdly

paolodamico committed:
animation & link

paolodamico committed:
large sidebar

paolodamico committed:
initial top navigation

paolodamico committed:
Merge branch 'master' into navigation-1775

macobo commented on frontend/src/scenes/App.tsx:
```diff
@@ -78,12 +81,22 @@ function _App(): JSX.Element {
                     style={{ minHeight: '100vh' }}
                 >
-                    <TopContent />
+                    {!featureFlags['navigation-1775'] && <TopContent />}
```
Nit: Using ternary would make sense here and elsewhere:

```
{featureFlags['navigation-1775'] ? <TopNavigation /> : <TopContent />}
```

paolodamico committed:
fix navigation bars

paolodamico committed:
mobile menu bar

paolodamico committed:
collapsible style

paolodamico committed:
random styles missing from #2114

paolodamico committed:
close navbar when item is selected on mobile

paolodamico committed:
style fixes

paolodamico committed:
top navigation

paolodamico committed:
top navbar badges

paolodamico committed:
add machine-friendly keys to system status metrics

paolodamico committed:
system status indicator

paolodamico committed:
update indicator

paolodamico committed:
right dropdown

paolodamico committed:
right dropdown functionality

paolodamico committed:
project chooser

paolodamico committed:
create new project

paolodamico committed:
switch projects & project settings

paolodamico committed:
fix eslint

paolodamico committed:
allow creating an organization

paolodamico committed:
actual nav items

paolodamico committed:
fix plugins page

paolodamico committed:
toolbar modal

paolodamico committed:
update husky

paolodamico committed:
Merge branch 'master' into navigation-1775

paolodamico committed:
show scroll indicator

yakkomajuri commented:
Awesome work @paolodamico! Will be adding feedback here as it comes. 

First thing - this is a bit annoying:

<img width="79" alt="Screenshot 2020-11-09 at 08 29 20" src="https://user-images.githubusercontent.com/38760734/98517408-c65da800-2265-11eb-86f5-c6d3e6361641.png">

Especially given that there's nothing down there

paolodamico committed:
minor adjustments

yakkomajuri commented on frontend/src/layout/navigation/navigationLogic.ts:
```diff
@@ -0,0 +1,111 @@
+                const aliveMetrics = ['redis_alive', 'db_alive']
+                let aliveSignals = 0
+                for (const metric of statusMetrics) {
+                    if (aliveMetrics.indexOf(metric.key) > -1 && metric.value) {
```
Nit: could use the more readable `aliveMetrics.includes(metric.key)`

yakkomajuri commented on frontend/src/lib/components/Badge.tsx:
```diff
@@ -0,0 +1,35 @@
+            return <WarningOutlined />
+        } else if (type === 'danger') {
+            return <WarningOutlined />
+        }
```
Nit: could become `return type === 'success' ? <CheckOutlined /> : <WarningOutlined />`

Or is there a reason this doesn't have a default return?

paolodamico replied:
Well the only reason for this would be to future-proof for more types of badges

yakkomajuri replied:
I mean, wouldn't be too complicated to add that logic then - but it's far from important to me to use the ternary operator here. 

The main question is about why not have a default return making it so that it may return `undefined`

paolodamico replied:
Oh I see your point, well because we do have some badges that have no icon (i.e. expected behavior), I'll add a note there

yakkomajuri commented on frontend/src/scenes/events/EventsActions.tsx:
```diff
@@ -0,0 +1,23 @@
+            </Tabs>
+        </>
+    )
+}
```
I'm guessing this is a mistake?

yakkomajuri reviewed (changes requested):
Some minor nits on the code side (non-blocking)  + a few issues on the UI side but otherwise this is great!

paolodamico committed:
fixes

paolodamico committed:
remove unused code

paolodamico committed:
fixes scroll indicator showing by default on large screens

paolodamico committed:
minor adjustments

yakkomajuri commented:
Would be nice if the changelog had an icon:

<img width="92" alt="Screenshot 2020-11-10 at 09 47 38" src="https://user-images.githubusercontent.com/38760734/98657471-d21a9e80-2339-11eb-9b9c-f5d248f744b4.png">


Also the project name is centered according to the entire screen whereas it feels off for me - my mind intuitively wants in centered on the panel i.e. not taking the sidebar into consideration.

If the top bar extended the entire screen it would make sense, but since it doesn't, I'd rather have it centered to the panel

mariusandra commented:
Looking good! Some small things I found:

1. Scrollbars! Using with an external mouse, the scrollbar is visible always on a Mac. It'll be especially visible for all windows users. Some `overflow: scroll` should be changed to `overflow: auto`.
2. "Feat. Flags" is not centred.
3. "Dashboards" text goes longer than the box.
<img width="948" alt="Screenshot 2020-11-12 at 13 28 19" src="https://user-images.githubusercontent.com/53387/98940333-1f8a3d80-24eb-11eb-8d67-8b49f5c812c2.png">

4. I expect that something happens when clicking the posthog logo... that I'm taken back to the "home" screen (whatever that is), but nothing happens when I click it.

5. If I'm under "my account", nothing in the menu nor in the header is highlighted. It might be nice to highlight the entire top-right corner to show that the page is from there:
![image](https://user-images.githubusercontent.com/53387/98940638-91fb1d80-24eb-11eb-9554-63245fe7e4e0.png)

6. There's some gray box (another scrollbar?) also there in mobile mode: 
![image](https://user-images.githubusercontent.com/53387/98940759-bf47cb80-24eb-11eb-8685-871a7c1b1c0a.png)

paolodamico committed:
update icon & center project chooser

paolodamico committed:
address more feedback

paolodamico commented:
Thanks for the review @yakkomajuri & @mariusandra! I've addressed all your comments, the only exception is @mariusandra's No. 5, looks a bit weird when doing this, but feel free to push back.

Feel free to review again, or if anyone else wants to do a more thorough code review, otherwise I'll merge before the code freeze.
